code/datums/status_effects/gun_skill.dm

```diff
@@ -4,12 +4,69 @@
 	delay_before_decay = 5 SECONDS
 	stack_threshold = 100
 	max_stacks = 100
+	///reference to particle effect holder is present for this stack, initially a reference to the type to use
+	var/obj/effect/abstract/particle_holder/particles = /particles/gun_skill
+
+/datum/status_effect/stacking/gun_skill/on_creation(mob/living/new_owner, stacks_to_apply)
+	. = ..()
+	if(!.)
+		return
+	particles = new(owner, particles)
+
+/datum/status_effect/stacking/gun_skill/add_stacks(stacks_added)
+	. = ..()
+	if(!.)
+		return
+	particles.particles.spawning = stacks
+
+/datum/status_effect/stacking/gun_skill/Destroy()
+	if(particles)
+		QDEL_NULL(particles)
+	return ..()
+
 
 /datum/status_effect/stacking/gun_skill/accuracy/buff
 	id = "gun_skill_accuracy_buff"
+	particles = /particles/gun_skill/accuracy/buff
+
 /datum/status_effect/stacking/gun_skill/accuracy/debuff
 	id = "gun_skill_accuracy_debuff"
+	particles = /particles/gun_skill/accuracy/debuff
+
 /datum/status_effect/stacking/gun_skill/scatter/buff
 	id = "gun_skill_scatter_buff"
+	particles = /particles/gun_skill/scatter/buff
+
 /datum/status_effect/stacking/gun_skill/scatter/debuff
 	id = "gun_skill_scatter_debuff"
+	particles = /particles/gun_skill/scatter/debuff
+
+/particles/gun_skill
```

Comment:
this effect is quite spammy IMO, maybe reduce lifespan or something.
![image](https://user-images.githubusercontent.com/7869430/203678836-14ea7b91-0ac3-47b8-abfb-d4945a606eda.png)

Reply:
It should be visible imo